bulbs/recirc/mixins.py

```diff
@@ -0,0 +1,115 @@
+from django.conf import settings
+from django.db import models
+
+from elasticsearch import Elasticsearch
+from json_field import JSONField
+
+from bulbs.content.custom_search import custom_search_model
+from bulbs.content.models import Content
+
+
+es = Elasticsearch(settings.ES_CONNECTIONS["default"]["hosts"])
+
+
+class InlineRecircMixin(models.Model):
+
+    query = JSONField(default={}, blank=True)
+
+    class Meta:
+        abstract = True
+
+    def clean(self):
+        super(InlineRecircMixin, self).clean()
+        self.clean_query()
+
+    def clean_query(self):
+        """
+        Removes any `None` value from an elasticsearch query.
+        """
+        if self.query and self.query != {}:
+            for key, value in self.query.items():
+                if isinstance(value, list) and None in value:
+                    self.query[key] = [v for v in value if v is not None]
+
+    def save(self, *args, **kwargs):
+        self.clean()
+
+        super(InlineRecircMixin, self).save(*args, **kwargs)
+
+        if self.query and self.query != {}:
+            self._save_percolator()
+
+    def _save_percolator(self):
```

Comment:
Do we need a percolator for this scenario?  With percolator, we use when we need to ask, is X item part of Y query.  But I didn't think we needed that scenario.  It's always give me a list of X items based on Y query